userpasswd.1
.\" Copyright (C) 1997 Red Hat Software, Inc.
.\"
.\" This is free software; you can redistribute it and/or modify it
.\" under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful, but
.\" WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
.\" General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License
.\" along with this program; if not, write to the Free Software
.\" Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
.\" MA  02110-1301, USA.
.\"
.TH USERPASSWD 1 "17 October 1997" "Red Hat Software"
.SH NAME
userpasswd \- A graphical tool to allow users to change their passwords.
.SH SYNOPSIS
.B userpasswd
[
.I options
]
.SH DESCRIPTION
.B userpasswd
is a graphical tool to allow users to easily change their passwords.
Aside from being a nice GUI frontend, this program does the right
thing when the pam configuration has changed.
.SH OPTIONS
This program has no command line options of its own, but it does take
the standard X program options like
.B -display
and such.  See the 
.IR X (1)
man page for some of the common options.
.SH "SEE ALSO"
.IR userhelper (8)
.SH BUGS
No known bugs, but compatibility with bizarre changes in the pam
configuration have not been tested aggressively.  If you have a
non-standard pam configuration, I'd be interested to hear whether it
worked and what kind of configuration you have.
.SH AUTHOR
Otto Hammersmith <otto@redhat.com>


NEWS
1.112:
* Updated translations.

1.111:
* Updated translations.

1.110:
* INCOMPATIBLE CHANGE: The shutdown(8) wrapper script and the
  $sysconfdir/security/console.apps/{halt,poweroff,reboot} configuration files
  are not installed by this package.  The contents have been moved to a
  "sample" subdirectory in the tarball.  Distributions are encouraged to
  package such wrapping configuration along with the wrapped commands instead,
  like they probably do for all other commmands.
* All commands now terminate on unexpected arguments intead of silently ignoring
  them.

1.109:
* Fix build with glib >= 2.30.
* Remove a misleading space in from syslog records.
* Updated translations.

1.108:
* Report errors in text-mode authentication to stderr.
* Fix a rare crash in usermount.
* Stop using deprecated GTK+ functions.

1.107:
* Minor bug fixes.
* A man page fix by John Bradshaw <john@johnbradshaw.org>
* Updated translations.

1.106.1:
* Update translations.

1.106:
* Update translations.
* Minor code cleanups.

1.105:
* Make startup notification work.
* Use a better error message if the user runs out of attempts allowed by PAM.

1.104.2:
* Update translations.

1.104.1:
* Update translations.

1.104:
* Fix silent exit in some non-UTF-8 locales instead of PAM interaction.

1.103:
* Supply TTY name to PAM.
* Use root's supplementary groups.
* Call pam_secred() (e.g. adds support for pam_group.so).
* Update translations.

1.102:
* Add more man pages for consolehelper-gtk and pam-panel-icon.
* Minor fixes in other man pages.
* Update translations.

1.101
* Use GtkBuilder instead of libglade.
* Avoid some obsolete GTK+ interfaces.
* Update translations.

1.100
* Don't use interfaces deprecated in GTK+ 2.14.

1.99
* Support multi-line prompts.

1.98.1
* Update translations

1.98
* Support dialogs with no text entries.
* Left-justify prompts.
* Remove /usr/X11R6/bin from default path.

1.97
* Fix display of '_' in prompts.

1.96.1
* Update translations

1.96
* Remove code that overrides SELinux contexts of processes started by
  userhelper.

1.95
* New home page at https://fedoraproject.org/usermode/ .
* Correctly preserve exit code when SESSION=yes
* Fix minor errors in the .desktop files

1.94
* Add support for including files from wrapper configuration files.  Original
  patch by Carlo de Wolf <cdewolf@redhat.com>.

1.93.1
* Allow building with startup notification support without libwnck
* Update translations

1.93
* Fix handling of PAM conversations with no questions

1.92
* Fix userhelper hangs in non-UTF8 locales.
* Fix unexpanded {prefix} strings embedded in executables if some directory
  variables were not explicitly passed to configure.
* Add menu categories to desktop files
* Minor clean-ups and optimizations.

1.91.2
* Show the user we're authenticating as in an i18n-safe way.

1.91.1
* Update translations

1.91
* Preserve environment variables in consolehelper specified in the service
  config file.
* Fix handling of empty pw_shell.
* Minor clean-ups and optimizations.

1.90
* Support LABEL= and UUID= in usermount and userformat.
* Fix an use of PAM data after free.
* Add a dialog caption when changing user password.
* Use less memory in pam-panel-icon until the icon is actually displayed.
* Don't use deprecated GTK+ widgets.
* Other minor bug fixes, clean-ups and optimizations.

1.89
* Preserve the exit status of processes executed by consolehelper.


userpasswd.c
/* -*-Mode: c-*- */
/* Copyright (C) 1997 Red Hat Software, Inc.
 * Copyright (C) 2001, 2007 Red Hat, Inc.
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
 * MA  02110-1301, USA.
 */

#include "config.h"
#include <stdio.h>
#include <stdlib.h>

#include <libintl.h>
#include <glib/gi18n.h>
#include <gtk/gtk.h>
#include "userhelper.h"
#include "userhelper-wrap.h"

int
main(int argc, char *argv[])
{
	bindtextdomain(PACKAGE, LOCALEDIR);
	bind_textdomain_codeset(PACKAGE, "UTF-8");
	textdomain(PACKAGE);

	gtk_init(&argc, &argv);

	if (argc != 1) {
		fprintf(stderr, _("Unexpected command-line arguments\n"));
		exit(1);
	}

	userhelper_run(TRUE, UH_PATH, UH_PATH, UH_PASSWD_OPT, 0);

	return 0;
}
